### DISCUSSION TITLE
Check failed: page_allocator->DiscardSystemPages( reinterpret_cast<void*>(memory_area.begin()), memory_area.size()).

### DISCUSSION THREAD

**Discussion by bittikettu** [nodejs/help#4948] [Q&A]
### Version

v18.3.0

### Platform

Linux exe4054-5181 5.4.78-rt44-exe1 #1 SMP Fri May 6 12:49:24 EEST 2022 armv7l GNU/Linux

### Subsystem

Own buildroot based kernel

### What steps will reproduce the bug?
Small app to read socket data. Data is coming quite fast in 24byte chunks.

`var net = require('node:net');

var client = new net.Socket();
client.connect(4028, '127.0.0.1', function() {
        console.log('Connected');
        //client.write('Hello, server! Love, Client.');
});

client.on('data', function(data) {
        console.log(data);
        //client.destroy(); // kill client after server's response
});

client.on('close', function() {
        console.log('Connection closed');
});`

### How often does it reproduce? Is there a required condition?

Issue occurs after start to listen socket. Few seconds go by and then the app crashes.

### What is the expected behavior?

Everything works fine with node-v16.15.0-linux-armv7l

### What do you see instead?

<Buffer 18 00 15 00 14 00 00 04 6c 00 64 00 48 2d ff 9b 18 0c 00 00 05 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 66 00 64 00 72 2d ff 9b 18 0c 00 00 06 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6b 00 64 00 87 2d ff 9b 18 0c 00 00 06 d2 2f 00 18 00 15 00 14 00 00 04 6a 00 64 00 87 2d ff 9b 18 0c 00 00 06 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 69 00 64 00 9c 2d ff 9b 18 0c 00 00 07 d2 2f 00 18 00 15 00 14 00 00 04 68 00 64 00 9c 2d ff 9b 18 0c 00 00 07 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6c 00 64 00 a7 2d ff 9b 18 0c 00 00 07 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 66 00 64 00 d1 2d ff 9b 18 0c 00 00 08 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6b 00 64 00 e5 2d ff 9b 18 0c 00 00 08 d2 2f 00 18 00 15 00 14 00 00 04 6a 00 64 00 e6 2d ff 9b 18 0c 00 00 08 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 69 00 64 00 05 2e ff 9b 18 0c 00 00 09 d2 2f 00 18 00 15 00 14 00 00 04 68 00 64 00 05 2e ff 9b 18 0c 00 00 09 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6c 00 64 00 0f 2e ff 9b 18 0c 00 00 09 d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 66 00 64 00 39 2e ff 9b 18 0c 00 00 0a d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6b 00 64 00 4e 2e ff 9b 18 0c 00 00 0a d2 2f 00 18 00 15 00 14 00 00 04 6a 00 64 00 4e 2e ff 9b 18 0c 00 00 0a d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 69 00 64 00 6d 2e ff 9b 18 0c 00 00 0b d2 2f 00 18 00 15 00 14 00 00 04 68 00 64 00 6d 2e ff 9b 18 0c 00 00 0b d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6c 00 64 00 78 2e ff 9b 18 0c 00 00 0b d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 66 00 64 00 97 2e ff 9b 18 0c 00 00 0c d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6b 00 64 00 ac 2e ff 9b 18 0c 00 00 0c d2 2f 00 18 00 15 00 14 00 00 04 6a 00 64 00 ac 2e ff 9b 18 0c 00 00 0c d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 69 00 64 00 cb 2e ff 9b 18 0c 00 00 0d d2 2f 00 18 00 15 00 14 00 00 04 68 00 64 00 cb 2e ff 9b 18 0c 00 00 0d d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 6c 00 64 00 d6 2e ff 9b 18 0c 00 00 0d d2 2f 00>
<Buffer 18 00 15 00 14 00 00 04 66 00 64 00 00 2f ff 9b 18 0c 00 00 0e d2 2f 00>

#
# Fatal error in , line 0
# Check failed: page_allocator->DiscardSystemPages( reinterpret_cast<void*>(memory_area.begin()), memory_area.size()).
#
#
#
#FailureMessage Object: 0xb49fb600
Bus error

**Comment by bnoordhuis** [comment]
Can you run strace on your program? I suspect there's a munmap/mprotect/madvise system call failing and I'd like to know with what error.

**Comment by bittikettu** [comment]
> Can you run strace on your program? I suspect there's a munmap/mprotect/madvise system call failing and I'd like to know with what error.

I'll run it right away.

**Comment by bittikettu** [comment]
`
read(18, "\30\0\25\0\24\0\0\4h\0d\0\210\316\24\234\30\f\0\0003@0\0\30\0\25\0\24\0\0\4"..., 65536) = 72
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=687590880}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=688818880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=689495880}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 272) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=706233213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=706727546}) = 0
read(18, "\30\0\25\0\24\0\0\4k\0d\0\307\316\24\234\30\f\0\0004@0\0\30\0\25\0\24\0\0\4"..., 65536) = 48
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=708434213}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=709571546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=709926546}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 251) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=726821213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=727333546}) = 0
read(18, "\30\0\25\0\24\0\0\4i\0d\0\334\316\24\234\30\f\0\0005@0\0", 65536) = 24
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=729245880}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=730500880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=730787546}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 230) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=789971880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=790729546}) = 0
read(18, "\30\0\25\0\24\0\0\4h\0d\0\361\316\24\234\30\f\0\0005@0\0\30\0\25\0\24\0\0\4"..., 65536) = 72
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=792684546}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=793619213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=794331213}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 167) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=800543546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=801097546}) = 0
read(18, "\30\0\25\0\24\0\0\4k\0d\0000\317\24\234\30\f\0\0006@0\0\30\0\25\0\24\0\0\4"..., 65536) = 48
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=803196546}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=804086213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=804632213}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 156) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=821527880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=821783880}) = 0
read(18, "\30\0\25\0\24\0\0\4i\0d\0:\317\24\234\30\f\0\0007@0\0", 65536) = 24
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=823636880}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=825094213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=825378880}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 135) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=883959880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=884475213}) = 0
read(18, "\30\0\25\0\24\0\0\4h\0d\0O\317\24\234\30\f\0\0007@0\0\30\0\25\0\24\0\0\4"..., 65536) = 72
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=886314546}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=887643213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=888143880}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 73) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=905836546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=906421546}) = 0
read(18, "\30\0\25\0\24\0\0\4k\0d\0\216\317\24\234\30\f\0\08@0\0\30\0\25\0\24\0\0\4"..., 65536) = 48
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=908464213}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=909091546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=909669546}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}], 1024, 51) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=926620213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=927188546}) = 0
read(18, "\30\0\25\0\24\0\0\4i\0d\0\243\317\24\234\30\f\0\09@0\0", 65536) = 24
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=929286546}) = 0
epoll_wait(13, [], 1024, 0)             = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=930389546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=930971546}) = 0
epoll_wait(13, [], 1024, 30)            = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=962003546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=962551880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=962845546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=963328213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=964001880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=964552213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=965108546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=965776546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=966343880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=967280880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=967799546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=968574213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=969110213}) = 0
futex(0x3f76fbc, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76fbc, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76fb8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76fb8, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=991790546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=993748880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=994265213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=995711880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4589, tv_nsec=996889213}) = 0
write(16, "\1\0\0\0\0\0\0\0", 8)        = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=99213}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}, {events=EPOLLIN, data={u32=16, u64=16}}], 1024, 0) = 2
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=4681880}) = 0
read(18, "\30\0\25\0\24\0\0\4h\0d\0\270\317\24\234\30\f\0\09@0\0\30\0\25\0\24\0\0\4"..., 65536) = 120
read(16, "\1\0\0\0\0\0\0\0", 1024)      = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=10406213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=10678546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=11437880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=12761546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=16085546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=17963213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=18186546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=18409213}) = 0
write(16, "\1\0\0\0\0\0\0\0", 8)        = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=18866546}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=18, u64=18}}, {events=EPOLLIN, data={u32=16, u64=16}}], 1024, 0) = 2
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=27737880}) = 0
read(18, "\30\0\25\0\24\0\0\4i\0d\0\2\320\24\234\30\f\0\0;@0\0", 65536) = 24
read(16, "\1\0\0\0\0\0\0\0", 1024)      = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=32616546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=32824213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=33004880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=33173546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=33334213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=37701880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=38854546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=39767546}) = 0
write(16, "\1\0\0\0\0\0\0\0", 8)        = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=41806213}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=16, u64=16}}], 1024, 0) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=44302213}) = 0
read(16, "\1\0\0\0\0\0\0\0", 1024)      = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=45434880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=45693880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=47117880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=47784880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=48248546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=48875880}) = 0
write(16, "\1\0\0\0\0\0\0\0", 8)        = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=50870546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=52190880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=52766213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=53322546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=53835880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=54079546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=54940213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=55212213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=55546546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=57100880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=57553546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=58317213}) = 0
epoll_wait(13, [{events=EPOLLIN, data={u32=16, u64=16}}], 1024, 0) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=60574546}) = 0
read(16, "\1\0\0\0\0\0\0\0", 1024)      = 8
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=61551213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=62454213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=63443546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=63999213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=64895213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=66339213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=66891546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=67533546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=68365546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=69617213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=70497880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=71931546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=72194546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=72857880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=73735546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=74402880}) = 0
mprotect(0x3d300000, 262144, PROT_READ|PROT_WRITE) = 0
mprotect(0x5b900000, 262144, PROT_READ|PROT_WRITE) = 0
mprotect(0x2d2c4000, 94208, PROT_READ|PROT_WRITE|PROT_EXEC) = 0
mprotect(0x267c4000, 241664, PROT_READ|PROT_WRITE|PROT_EXEC) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=77298880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=77959213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=78735213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=79316880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=79973880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=80745546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=81816213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=82063213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=82606880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=83574880}) = 0
futex(0x3f76fbc, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1) = 1
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=85614546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=86303546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=86956546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=87766546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=88311546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=88634213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=89221546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=89711213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=90284880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=90887880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=91096880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=91656546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=92297880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=92970546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=93411546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=94072546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=94329546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=94904880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=95565213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=95780880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=97266213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=97997546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=98360880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=99048546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=99385213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=100209546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=100826213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=101215213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=101922546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=102338546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=103299546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=104593880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=105192546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=105752880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=106348546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=106938213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=107230213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=107791880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=108501213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=109042213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=109633546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=110237213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=110508213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=111351213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=111603880}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=112140546}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=112884213}) = 0
clock_gettime64(CLOCK_MONOTONIC, {tv_sec=4590, tv_nsec=113163546}) = 0
futex(0x3f76fbc, FUTEX_WAKE_PRIVATE, 1) = 1
futex(0x3f76f78, FUTEX_WAKE_PRIVATE, 1

#
# Fatal error in , line 0
# Check failed: page_allocator->DiscardSystemPages( reinterpret_cast<void*>(memory_area.begin()), memory_area.size()).
#
#
#
#FailureMessage Object: 0xb4afb600
) = ?
+++ killed by SIGBUS +++
Bus error
`

**Comment by bnoordhuis** [comment]
Can you run `strace -fo trace.log node app.js`? That captures all threads.

**Comment by bittikettu** [comment]
Just a side notice, while running it with strace the error didn't occur that fast.

**Comment by bittikettu** [comment]
> Can you run `strace -fo trace.log node app.js`? That captures all threads.

Seems to be not crashing while using strace.

**Comment by bittikettu** [comment]
YAS!

Had to took the filesystem writing off. The amount of logs pushing to the flash in our embedded device slightly caused a bottleneck, but I was able to get it to trigger when printing to console.

[strace.log](https://github.com/nodejs/node/files/8916771/strace.log)

**Comment by bittikettu** [comment]
Hmm, socket read turned into socket write.

**Comment by bnoordhuis** [comment]
Looks like your kernel is broken:
```
[pid 15120] madvise(0x27710000, 196608, MADV_DONTNEED <unfinished ...>
[..]
[pid 15120] <... madvise resumed>)      = -1 ENOSYS (Function not implemented)
```
edit: converted to a discussion because it's not a node bug.

**Reply by bittikettu** [reply]
Thank you!
